src/Mod/Fem/Gui/ViewProviderSolver.cpp

```diff
@@ -55,11 +59,29 @@ std::vector<std::string> ViewProviderSolver::getDisplayModes() const
 
 bool ViewProviderSolver::onDelete(const std::vector<std::string>&)
 {
-    // warn the user if the object has childs
+    // warn the user if the object has unselected childs
 
     auto objs = claimChildren();
-    if (!objs.empty())
-    {
+    if (!objs.empty()) {
+        // check if all childs are in the selection
+        bool found = false;
+        auto docGui = Gui::Application::Instance->activeDocument();
+        auto selectionList = Gui::Selection().getSelectionEx(docGui->getDocument()->getName());
+        for (auto child : objs) {
+            found = false;
+            for (Gui::SelectionObject selection : selectionList) {
+                if (std::string(child->getNameInDocument())
+                    == std::string(selection.getFeatName())) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found)
+                break;
+        }
+        if (found)// all childs are selected too
```

Comment:
According to dictionaries yes, but in real-life it is quite common (I read it even in scientific papers). I also think we can be more flexible since people understand the comment and "childs" is shorter than "children". (language lives ;-) )